lib/msf/core/exploit/http/client.rb

```diff
@@ -508,6 +508,31 @@ def path_from_uri(uri)
     end
   end
 
+  #
+  # Returns a hash of request opts from a URL string
+  def request_opts_from_url(url)
+    tgt = URI.parse(url)
+    opts = { 'rhost' => tgt.host, 'rport' => tgt.port, 'uri' => tgt.request_uri }
+    opts['SSL'] = true if tgt.scheme == 'https'
+    if tgt.query and tgt.query.size > 13
+      # Assming that this is going to be mostly used for GET requests as string -> req
+      opts['vars_get'] = {}
+      tgt.query.split('&').each do |pair|
+        k,v = pair.split('=',2)
+        opts['vars_get'][k] = v
+      end
+    end
+    return opts
+  end
+
+  #
+  # Returns response from a simple URL call
+  def request_url(url, keepalive = false)
+    res = send_request_raw(request_options_from_url(url))
```

Comment:
Yeah, OK. This is why I may not be ideally suited to working at the interface level. Wilco